bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/ui/QuickSearchDialog.java

```diff
@@ -811,6 +791,7 @@ public void getName(AccessibleEvent e) {
 
 		pattern.addModifyListener(e -> {
 			applyFilter(false);
+			searchIn.storeHistory();
```

Comment:
on every character typed?

Reply:
Yes, because every new character typed starts a new search. Storing history is cheap, so I don't think that's a problem

Reply:
compare to https://github.com/eclipse-platform/eclipse.platform.ui/issues/2291

Reply:
I believe you refer to
> Since the overlay has incremental search enabled by default (other than the existing dialog), adding every searched term to the history is inappropriate, as typing "test" would result in history entries "t", "te", "tes" and "test".

However, here we are doing something slightly differently: the search history is applied to the "search in" field - and we store the "search in" history every time some search is performed - while typing in the search bar, there is no change of the "search in" field. Thus, we save the same string to history for every keystroke. However, since we also remove all other occurrences of that exact string in the history, the operation does nothing.

Anyway, I don't want to invest too much into this PR. It is a good model for a new contributor wanting to play with this issue.

The main problems are the dependencies; I am not so sure about importing two internal packages

Reply:
please mark as draft